src/main/java/io/cdap/plugin/google/sheets/sink/GoogleSheetsSinkConfig.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.google.sheets.sink;
+
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.plugin.google.common.GoogleInputSchemaFieldsUsageConfig;
+
+import java.util.Arrays;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * Configurations for Google Sheets Batch Sink plugin.
+ */
+public class GoogleSheetsSinkConfig extends GoogleInputSchemaFieldsUsageConfig {
+  public static final String SHEET_NAME_FIELD_NAME = "sheetName";
+  public static final String SPREADSHEET_NAME_FIELD_NAME = "spreadSheetName";
+  public static final String SCHEMA_SPREAD_SHEET_NAME_FIELD_NAME = "schemaSpreadSheetNameFieldName";
+  public static final String SCHEMA_SHEET_NAME_FIELD_NAME = "schemaSheetNameFieldName";
+  public static final String WRITE_SCHEMA_FIELD_NAME = "writeSchema";
+  public static final String MERGE_DATA_CELLS_FIELD_NAME = "mergeDataCells";
+  public static final String MIN_PAGE_EXTENSION_PAGE_FIELD_NAME = "minPageExtensionSize";
+  public static final String THREADS_NUMBER_FIELD_NAME = "threadsNumber";
+  public static final String MAX_BUFFER_SIZE_FIELD_NAME = "maxBufferSize";
+  public static final String RECORDS_QUEUE_LENGTH_FIELD_NAME = "recordsQueueLength";
+  public static final String MAX_FLUSH_INTERVAL_FIELD_NAME = "maxFlushInterval";
+  public static final String FLUSH_EXECUTION_TIMEOUT_FIELD_NAME = "flushExecutionTimeout";
+
+  public static final String TOP_LEVEL_SCHEMA_MESSAGE =
+    "Field '%s' has unsupported schema type '%s' and logical type '%s'.";
+  public static final String TOP_LEVEL_SCHEMA_CORRECTIVE_MESSAGE =
+    "Allowed top level logical types: [%s] and types: [%s].";
+  public static final String ARRAY_NESTED_SCHEMA_MESSAGE =
+    "Array field '%s' has unsupported components' schema type '%s' and logical type '%s'.";
+  public static final String ARRAY_NESTED_SCHEMA_CORRECTIVE_MESSAGE =
+    "Allowed array nested logical types: [%s] and types: [%s].";
```

Comment:
Supported array types: '%s' and logical types: '%s'.

Reply:
done